crates/bevy_dev_tools/src/ci_testing.rs

```diff
@@ -20,30 +19,26 @@ use serde::Deserialize;
 /// exit a Bevy app when run through the CI. This is needed because otherwise
 /// Bevy apps would be stuck in the game loop and wouldn't allow the CI to progress.
 #[derive(Deserialize, Resource)]
-pub struct CiTestingConfig {
-    /// The number of frames after which Bevy should exit.
-    pub exit_after: Option<u32>,
-    /// The time in seconds to update for each frame.
-    pub frame_time: Option<f32>,
-    /// Frames at which to capture a screenshot.
+struct CiTestingConfig {
+    /// The setup for this test.
+    #[serde(default)]
+    setup: CiTestingSetup,
+    /// Events to send, with their associated frame.
     #[serde(default)]
-    pub screenshot_frames: Vec<u32>,
+    events: Vec<CiTestingEvent>,
 }
 
-fn ci_testing_exit_after(
-    mut current_frame: bevy_ecs::prelude::Local<u32>,
-    ci_testing_config: bevy_ecs::prelude::Res<CiTestingConfig>,
-    mut app_exit_events: bevy_ecs::event::EventWriter<AppExit>,
-) {
-    if let Some(exit_after) = ci_testing_config.exit_after {
-        if *current_frame > exit_after {
-            app_exit_events.send(AppExit::Success);
-            info!("Exiting after {} frames. Test successful!", exit_after);
-        }
-    }
-    *current_frame += 1;
+/// Setup for a test.
+#[derive(Deserialize, Default)]
+struct CiTestingSetup {
+    /// The time in seconds to update for each frame.
+    pub frame_time: Option<f32>,
```

Comment:
Could you be a bit more specific about what `frame_time` is actually used for? (Maybe reference `TimeUpdateStrategy`?)